src/service.rs

```diff
@@ -57,6 +57,22 @@ pub struct BlockInfo<Types: NodeType> {
     pub offered_fee: u64,
 }
 
+#[derive(Debug)]
+pub struct BuiderStatesInfo<Types: NodeType> {
+    pub vid_commitments: Vec<VidCommitment>,
+    pub block_ids: Vec<(VidCommitment, BuilderCommitment, Types::Time)>,
```

Comment:
Comment would be helpful here also, particularly I'm wondering why we need both `VidCommitment` and `BuilderCommitment`. I think it's because `VidCommitment` (together with `Types::Time`) is uniquely identifying the parent, while the `BuilderCommitment` is uniquely identifying the proposed block, as we may have built several alternative blocks on the same parent. I'm not sure though. If that interpretation is right might I suggest another struct to replace the remaining tuple:

```rust
struct ProposedBlockId {
  parent_commitment: VidCommitment,
  parent_view: Types::Time,
  payload_commitment: BuilderCommitment,
}
```

Reply:
https://github.com/EspressoSystems/hotshot-builder-core/pull/120/commits/1fe4bfb9ee3497fa449375ba08adebf788a5a04a